implementations/rust/ockam/ockam_api/src/cli_state/journeys/attributes.rs

```diff
@@ -99,6 +100,17 @@ pub(crate) fn make_host() -> String {
     convert_to_hex(&hash(host))
 }
 
+/// Fill or trim a string so that it makes exactly the desired size
+fn adjust(s: String, desired_size: usize, filler: char) -> String {
+    let mut result = s;
+    let current_size = result.len();
+    // make sure that there exactly the desired number of characters
+    if current_size < desired_size {
+        result.extend(std::iter::repeat(filler).take(desired_size - current_size));
+    };
+    result[0..25].to_string()
+}
```

Comment:
Thanks @adrianbenavides, that's so much better than me reinventing the wheel!

Reply:
@adrianbenavides I eventually reverted to the original code because the string can be filled in some cases but trimmed in others and `format!` does not seem to support this use case.